### PR TITLE
Feature/blog article page

### LINKED ISSUE #80: Create and style the blog article page based on finished designs
## Description

Do all the necessary layout and styling work for the blog article page. The latest design is [here](https://www.figma.com/file/e2FveRnPmCidpGnuyTWbGj/Kedro-Blog-Ideas?node-id=8%3A51336&t=gy7XoVAR83uTeR6L-1).

### Acceptance criteria

- [x] Hero
- [x] Body text and headers
- [x] Sticky side navigation
- [x] Code highlighting (inline and block)
- [x] Blockquote
- [x] Media (image/video)
- [x] Conclusion
- [x] Author details
- [ ] Related posts (maybe?)

### PR DESCRIPTION
**Pull request by tynandebold**
## Description

Resolves #80.

## Development notes

This creates and styles the blog post page from the ground up.

It's quite a large PR. Sorry about that. Thankfully nothing in here is too complicated, code wise, so it should be easy enough to review, even though there are a lot of changes. Happy to pair on the review if it's easier.

## QA notes

[Take a look at the deployed Netlify URL](https://deploy-preview-81--kedro-website.netlify.app/blog/what-is-kedro-viz-2) and check it against the designs.

_Note_: instead of the design's **Related posts** at the bottom of the page, I've added **All blog posts** instead, since we don't have enough posts or content for a related-posts area.

I'm still missing a some small things:

- [x] Back to blog home icon (need from @Mackay031)
- [ ] Design guidance on what should happen when a user clicks **Copy link** (need from @Mackay031)
- [ ] Better UX behavior with the sticky nav. There was no design spec for that, so for now, it's a minimal implementation. (need from @Mackay031 if we want to change from how it is now)

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Added tests to cover my changes, where applicable

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -1,22 +1,26 @@
 const POST_GRAPHQL_FIELDS = `
 sys {
+  firstPublishedAt
```
Comment (rashidakanchwala) [inline lib/api.js:3]: just curious - what is the difference between firstPublishedAt and publishedAt. i thot it would be publishedAt and updatedAt?

**Review by rashidakanchwala — APPROVED** [review]
Thanks!! looks amazing :D

File: lib/api.js (hunk 1 above)
Comment (tynandebold) [reply lib/api.js:3]: Good question. What we're using here are Contentful's naming conventions. Search for `publishedAt` on [this page](https://www.contentful.com/developers/docs/references/content-management-api/#/introduction/collection-resources-and-pagination) to see the explanations.

**Comment by astrojuanlu** [comment]
Is having an Atom or RSS feed in scope for this PR?

**Review by Mackay031 — APPROVED** [review]

**Comment by tynandebold** [comment]
No. That isn't something we've even discussed. We should explore it.

For context, the acceptance criteria for this PR are [here](https://github.com/kedro-org/kedro-website/issues/80).

**Pull request merged**